Sources/FluidAudio/TextToSpeech/Kokoro/Assets/Lexicon/LexiconAssetManager.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+import OSLog
+
+@available(macOS 13.0, iOS 16.0, *)
+actor LexiconAssetManager {
+    static let shared = LexiconAssetManager()
+
+    private let logger = AppLogger(subsystem: "com.fluidaudio.tts", category: "LexiconAssetManager")
+    private let baseURL = "https://huggingface.co/FluidInference/kokoro-82m-coreml/resolve/main"
+
+    private func downloadFileIfNeeded(filename: String, urlPath: String) async throws {
```

Comment:
Hmm the downloading logics are a bit confusing, theres download utils, extension to that, voice emebdding and now assets too? 

Also all the files under Assets/ are in the /Lexicon folder?